bot/cogs/grades_data/grades_cog.py

```diff
@@ -234,6 +242,59 @@ def searchCourse(self, query, year):
     def go(self, course, year):
         return self.searchCourse(course, year)
 
+    def get_professor_query(self, prof_name, detailed):
+        # print("NAME: " + prof_name)
+        primary = self.global_master_prof_list[prof_name]
+        courses = {}
+        
+        for section in primary:
+            course = section['course'] + "-" + section['number']
+            courses[course] = {
+                "A": [],
+                "B": [],
+                "C": [],
+                "D": [],
+                "F": [],
+                "W": []
+            }
+        for section in primary:
```

Comment:
why are we iterating primary twice one after another?

Reply:
initialization of the dictionaries vs actually appending values to them

Reply:
you dont need to iterate twice for that

Reply:
I will anyway. They're lists of 10 at the most. No need to worry about it. Might look bad but other than that, it's fine